ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlockValidator.java

```diff
@@ -88,46 +84,44 @@ public BlockValidator(RecentChainData recentChainData, StateTransition stateTran
     }
 
     return recentChainData
-        .retrieveBlockState(block.getMessage().getParentRoot())
-        .thenApply(
-            preState -> {
-              if (preState.isEmpty()) {
+        .retrieveBlockByRoot(block.getMessage().getParentRoot())
+        .thenCompose(
+            parentBlock -> {
+              if (parentBlock.isEmpty()) {
                 LOG.trace(
-                    "BlockValidator: Block pre state does not exist. It will be saved for future processing");
-                return InternalValidationResult.SAVE_FOR_FUTURE;
+                    "BlockValidator: Parent block does not exist. It will be saved for future processing");
```

Comment:
while we're here, is it worth including block root in the messages for context?

Reply:
It's a trace level message so I'm not worried about it - mostly useful when a test is failing and you need to know why your block is being rejected.